ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Util/Enclose.hs

```diff
@@ -32,3 +36,18 @@ encloseWith ::
   -> m a
 encloseWith tracer action =
     traceWith tracer RisingEdge *> action <* traceWith tracer FallingEdge
+
+type EnclosingTimed = Enclosing' Word64
+
+encloseTimedWith ::
+     MonadMonotonicTimeNSec m
+  => Tracer m EnclosingTimed
+  -> m a
+  -> m a
+encloseTimedWith tracer action = do
+    before <- getMonotonicTimeNSec
+    traceWith tracer RisingEdge
+    res <- action
+    after <- getMonotonicTimeNSec
+    traceWith tracer (FallingEdgeWith (after - before))
+    pure res
```

Comment:
Maybe rather use [`getMonotonicTime`](https://hackage.haskell.org/package/si-timers-1.5.0.0/docs/Control-Monad-Class-MonadTime-SI.html#v:getMonotonicTime) and [`diffTime`](https://hackage.haskell.org/package/si-timers-1.5.0.0/docs/Control-Monad-Class-MonadTime-SI.html#v:diffTime) for more descriptive types instead of `Word64`?

Reply:
Fair, will do